server/src/main/java/org/opentosca/toscana/plugins/cloudformation/mapper/CapabilityMapper.java

```diff
@@ -0,0 +1,310 @@
+     */
+    public String mapComputeCapabilityToInstanceType(ComputeCapability computeCapability, String distinction) throws
+        TransformationFailureException {
+        //TODO what to do with disk size?
+        Integer numCpus = computeCapability.getNumCpus().orElse(0);
+        Integer memSize = computeCapability.getMemSizeInMB().orElse(0);
+        //default type the smallest
+        final ImmutableList<InstanceType> instanceTypes;
+        if ("EC2".equals(distinction)) {
+            instanceTypes = EC2_INSTANCE_TYPES;
+        } else if ("RDS".equals(distinction)) {
+            instanceTypes = RDS_INSTANCE_CLASSES;
+        } else {
+            throw new IllegalArgumentException("Distinguisher not supported");
+        }
+        List<Integer> allNumCpus = instanceTypes.stream()
+            .map(InstanceType::getNumCpus)
+            .sorted()
+            .collect(Collectors.toList());
+        List<Integer> allMemSizes = instanceTypes.stream()
+            .map(InstanceType::getMemSize)
+            .sorted()
+            .collect(Collectors.toList());
+        // scale numCpus and memSize upwards if they are not represented in the lists
+        try {
+            logger.debug("Check numCpus: " + numCpus);
+            numCpus = checkValue(numCpus, allNumCpus);
+            logger.debug("Check memSize: " + memSize);
+            memSize = checkValue(memSize, allMemSizes);
+        } catch (IllegalArgumentException ie) {
+            String errorMessage = "Values numCpus: " + numCpus + " and/or memSize: " + memSize + " are to big. No " +
+                "InstanceType found";
+            logger.error(errorMessage);
+            throw new TransformationFailureException(errorMessage, ie);
+        }
+        //get instanceType from combination
+        String instanceType = findCombination(numCpus, memSize, instanceTypes, allNumCpus, allMemSizes);
+        logger.debug("InstanceType is: " + instanceType);
+        return instanceType;
+    }
+
+    /**
+     The value is taken from the Capability but turned into GB. The minimum is 20 GB the maximum 6144 GB
+
+     @param computeCapability The ComputeCapability to map.
+     @return An integer representing the diskSize that should be taken.
+     */
+    public Integer mapComputeCapabilityToDiskSize(ComputeCapability computeCapability) {
+        final Integer minSize = 20;
+        final Integer maxSize = 6144;
+        Integer diskSize = computeCapability.getDiskSizeInMB().orElse(minSize * 1000);
+        diskSize = diskSize / 1000;
+        if (diskSize > maxSize) {
+            logger.debug("Disk size: " + maxSize);
+            return maxSize;
+        }
+        if (diskSize < minSize) {
+            logger.debug("Disk size: " + minSize);
+            return minSize;
+        }
+        logger.debug("Disk size: " + diskSize);
+        return diskSize;
+    }
+
+    private Integer checkValue(Integer value, List<Integer> checker) throws IllegalArgumentException {
```

Comment:
You should probably add comments or change the name for the following two methods, "checkValue" and "findCombination" don't tell me anything about their functionality.